File: lib_api/src/main/java/com/example/lib/api/CustomRecyclerView.java
package com.example.lib.api;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.ViewConfiguration;

import androidx.recyclerview.widget.RecyclerView;

public class CustomRecyclerView extends RecyclerView {
    private float initialY; // 记录初始Y坐标

    public CustomRecyclerView(Context context) {
        super(context);
    }

    public CustomRecyclerView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomRecyclerView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        // 当手指按下时记录初始Y坐标
        if (ev.getAction() == MotionEvent.ACTION_DOWN) {
            initialY = ev.getY();
        }

        // 判断当前滑动是否为上下滑动
        if (ev.getAction() == MotionEvent.ACTION_MOVE) {
            float deltaY = ev.getY() - initialY;
            //getScaledTouchSlop()当大于这个距离时才会有滑动事件
            if (Math.abs(deltaY) > ViewConfiguration.get(getContext()).getScaledTouchSlop()) {
                // 上下滑动，请求父级不要拦截事件
                getParent().requestDisallowInterceptTouchEvent(true);
            }
        }

        return super.dispatchTouchEvent(ev);
    }
}

File: model_search/src/main/java/com/example/model/search/CustomRecyclerView.kt
package com.example.model.search

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.recyclerview.widget.RecyclerView
import java.lang.Math.abs

class CustomRecyclerView(context: Context, attrs: AttributeSet?) : RecyclerView(context, attrs) {
    private var startX: Float = 0f
    private var startY: Float = 0f

    override fun onInterceptTouchEvent(e: MotionEvent): Boolean {
        when (e.action) {
            MotionEvent.ACTION_DOWN -> {
                startX = e.x
                startY = e.y
                parent.requestDisallowInterceptTouchEvent(true) // 禁止父布局拦截触摸事件
            }
            MotionEvent.ACTION_MOVE -> {
                val dx = abs(e.x - startX)
                val dy = abs(e.y - startY)
                if (dx > dy) {
                    parent.requestDisallowInterceptTouchEvent(true)
                } else {
                    parent.requestDisallowInterceptTouchEvent(false)
                }
            }
        }
        return super.onInterceptTouchEvent(e)
    }
}
